HW0/P5/P5_6.py

```diff
@@ -0,0 +1,82 @@
+# -*- coding: utf-8 -*-
+"""
```

Comment:
Was this necessary? I applaud your dedication and commitment, but doesn't this seem redundant given that we already have a mathematical solution derived in P5? Maybe I'm misreading this, but couldn't all of this be expressed with the one-liner `ceil(log_2(N))`?